x-pack/solutions/observability/plugins/apm/server/routes/traces/get_unified_trace_items.ts

```diff
@@ -116,7 +130,11 @@ export async function getUnifiedTraceItems({
           must: [
             {
               bool: {
-                filter: [...termQuery(TRACE_ID, traceId), ...rangeQuery(start, end)],
+                filter: [
+                  ...termQuery(TRACE_ID, traceId),
+                  ...rangeQuery(start, end),
+                  ...serviceNameQuery,
```

Comment:
I think you can define the service name query here directly:
```suggestion
                  ...(serviceName ? [termQuery(SERVICE_NAME, serviceName)] : []),
```

Reply:
Maybe a better name (`optionalServiceNameQuery`?) is better here? I find that the inline ternary pattern is really visually/syntactically noisy and not all that helpful in some cases (and with some of these objects, they are pretty big already so we should be thinking about how to reduce visual/syntax noise).